r2dbc-mysql/src/main/java/ExtendedStatement.kt

```diff
@@ -5,15 +5,17 @@ import io.r2dbc.spi.Result
 import io.r2dbc.spi.Statement
 import org.reactivestreams.Publisher
 import reactor.core.publisher.Mono
+import reactor.core.publisher.toFlux
 import reactor.core.publisher.toMono
-import java.util.concurrent.CompletableFuture
 import java.util.function.Supplier
 import com.github.jasync.sql.db.Connection as JasyncConnection
 
-class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, private val sql: String) : Statement {
```

Comment:
I think calling it StatementImpl might be more appropriate (or JasyncStatement)

Reply:
👍 for `JasyncStatement`